docs/config/01-configuration-file.md

```diff
@@ -67,6 +67,69 @@ require('ts-node').register({
 require('./karma.conf.ts');
 ```
 
+### Asynchronous Configuration
+
+The function exported by the configuration file may return a promise whose
+resolution will tell Karma when all configuration is complete, including the
+final call to `config.set()`.
+
+Just as with a synchronous configuration's return value not being used, an
+asynchronous configuration's resolved value will not be used. With both
+configuration types, `config.set()` MUST BE called to assign the configuration
+object.
+
+When using the command line interface, returning a promise is all that is
+needed.
+
+
+#### Asynchronous Configuration Using The Public API in Karma 6.x
```

Comment:
I don't think it is worth having this section here for two reasons:

- It will work out of the box in the upcoming v7.
- This file is primarily intended for the non-programmatic Karma users and programmatic Karma users will use the Public API page anyway where the concept is described in much detail already.

Reply:
I'd still like to point newer users in the right direction (just in case they don't think to check the API docs.)

How does this compromise look?

```diff
 ### Asynchronous Configuration

 The function exported by the configuration file may return a promise whose
 resolution will tell Karma when all configuration is complete, including the
 final call to `config.set()`.

 Just as with a synchronous configuration's return value not being used, an
 asynchronous configuration's resolved value will not be used. With both
 configuration types, `config.set()` MUST BE called to assign the configuration
 object.

 When using the command line interface, returning a promise is all that is
 needed.
-
-
-#### Asynchronous Configuration Using The Public API in Karma 6.x
-
-The ability to support promises was added as a feature in Karma `6.3.0`, but
-supporting it by default in the public API would be a breaking change. While
-this may change may in a future major version, users of Karma 6's public API
-have a few extra steps to go through in order to use promises. This way, the 
-behavior of the API does not change unexpectedly for existing users of versions
-`^6.3.0`.
-
-```js
-const path = require('path');
-
-const karma = require('karma');
-
-const cfg = karma.config;
-const Server = karma.Server;
-
-// Read karma.conf.js and start the Karma server
-cfg.parseConfig(
-  // **configFilePath** the file system path to the config file.
-  path.resolve('./karma.conf.js'),
-
-  // **cliOptions** takes priority over anything set by the config file.
-  {},
-
-  // **parseOptions** here we tell parseConfig that we want to use promises.
-  // Note that `throwErrors` MUST also be true to get promise rejections in
-  // place of immediately exiting the process.
-  { promiseConfig: true, throwErrors: true }
-).then(
-  (karmaConfig) => {
-    // use the config with the public API
-    const server = new Server(karmaConfig, function doneCallback(exitCode) {
-      console.log('Karma has exited with ' + exitCode);
-      process.exit(exitCode);
-    })
-  },
-  (rejectReason) => { 
-    // respond to the rejection reason error
-    console.log(rejectReason);
-    process.exit(1);
-  }
-);
-```
-
-For more information on using the public API, please see the
-[Public API Documentation][dev/public-api]]
+
+ If you are using the public API, then please see the 
+[Public API Documentation][dev/public-api]] for details, usage, and addtional
+information.
```

Hopefully it won't automatically date itself in v7+

Reply:
Sounds good, thanks!